Use client with static client_id in datalake throttling test

mmaslankaprv opened the pull request:
Datalake throttling uses the client_id to verify if ProduceRequest may
contain topics which are iceberg enabled. If the producer previously
produced to the iceberg enabled throttling it may be doing it again and
it will be throttled. Using static client id makes the test robust.

Fixes: CORE-9671
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/64690-->
<!--287d979f-a3c1-4ace-81ac-c076f48da855-->
<!--c0926ba3-22a0-4ac4-86fe-9a1fb4fccdf7-->
<!--title-->
## Retry command for [Build#64690](https://buildkite.com/redpanda/redpanda/builds/64690)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/upgrade_test.py::UpgradeBackToBackTest.test_upgrade_with_all_workloads@{"single_upgrade":false}
tests/rptest/tests/datalake/cluster_restore_test.py::DatalakeClusterRestoreTest.test_restore_partition_spec@{"catalog_type":"rest_hadoop","cloud_storage_type":1}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--ea9e19ad-5da5-4586-bece-4eb51f5a1c41-->
<!--TEST RESULTS-->
<!--94dfd992-0676-49f4-9a5b-f08fd459364d-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/64690">build#64690</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/64690#019644e5-131b-4302-9c3a-f4de62bfc30c | FLAKY | 19/21 |
| rptest.tests.upgrade_test.UpgradeBackToBackTest.test_upgrade_with_all_workloads.single_upgrade=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/64690#019644e1-48cc-429b-a593-495d5de3251e | FLAKY | 14/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/64739">build#64739</a></summary>


| test_id                                                                                              | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-----------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.upgrade_test.UpgradeBackToBackTest.test_upgrade_with_all_workloads.single_upgrade=False | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64739#01964877-1367-4733-9b48-2fc56d9c8042 | FLAKY         | 14/21    |
</details>
<!--/body-->

mmaslankaprv commented:
/ci-repeat 4
release
skip-redpanda-build
dt-repeat=10
tests/rptest/tests/datalake/throttling_test.py::DatalakeThrottlingTest

nvartolomei commented on tests/rptest/tests/datalake/throttling_test.py:
```diff
@@ -73,7 +73,9 @@ def _throttled_requests(self):
-        dl.produce_to_topic(self.topic_name, 128, 10240)
+        dl.produce_to_topic(self.topic_name, 1024, 3 * 10240)
+        rpk = RpkTool(self.redpanda)
+        rpk.produce(self.topic_name, "test_key", "test_value")
```
is this the only commit? rpk here sends a single produce request, how does that get throttled while kgo verifier doesn't? a comment explaining what's going on there would be helpful

the other question is more broad: in the datalake context, why client id is important in throttling? i would have assumed throttling should be per partition depending on load, no?

mmaslankaprv replied:
The client id is used to determine if the producer is the one that produces to Iceberg enabled topic. 
The per partition throttling would be ideal but the problem with this approach is that we would require decoding the whole produce request in memory. The throttling is applied after the request header is decoded. This is why we track the iceberg related producers using its client id. The producer is marked as iceberg related after the produce request has been processed.

nvartolomei replied:
ah i see why i think, you are calling this in some sort of a loop so rpk actually is invoked multiple times

a code comment would still help

i'd also consider not using produce_to_topic at all and call directly

```
KgoVerifierProducer.oneshot(self.test_ctx,
                                    self.redpanda,
                                    topic,
                                    msg_size=msg_size,
                                    msg_count=msg_count,
                                    rate_limit_bps=rate_limit_bps)
```

this is the only thing produce_to_topic does

AND add support for static name with a `client_name` which is passed to kgo instead of `self.who_am_i()`

explicit, cleaner, clearer—no need to break your brain understanding what's going on there

mmaslankaprv replied:
will do it

mmaslankaprv committed:
test/kgo: added ability to explicitly set producer id

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests/dl: use static client_id in datalake throttling test

Datalake throttling uses the client_id to verify if ProduceRequest may
contain topics which are iceberg enabled. If the producer previously
produced to the iceberg enabled throttling it may be doing it again and
it will be throttled. Using static client id makes the test robust.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/64739-->
<!--4da92e39-1efc-4fa4-97e1-d921c6efa617-->
<!--title-->
## Retry command for [Build#64739](https://buildkite.com/redpanda/redpanda/builds/64739)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/upgrade_test.py::UpgradeBackToBackTest.test_upgrade_with_all_workloads@{"single_upgrade":false}
```
<!--/body-->

vbotbuildovich commented:
/backport v25.1.x
